authentication/views.py

```diff
@@ -1,13 +1,52 @@
-from django.shortcuts import render
+from django.shortcuts import render,redirect 
 from django.contrib.auth import login,logout,authenticate
+from django.contrib.auth.forms import UserCreationForm
+from django.contrib.auth.models import auth,User
+from django.http import JsonResponse,HttpResponseRedirect
+from django.contrib import messages
 # Create your views here.
 
 
 def loginView(request):
-    pass
-
+    if request.method=='POST':
+        username = request.POST.get('Username')
+        password = request.POST.get('password')
+        user = auth.authenticate(username=username,password=password)
+        if user is not None:
+            auth.login(request,user)
+            contex={
+                'message':'success'
+            } 
+            return JsonResponse(contex)
+        else:
+            contex={
+                'message':'failed'
+            }
+            return JsonResponse(contex)
+    else:
+        return render(request,'registration/login.html')
 def logoutView(request):
     pass
 
 def registerView(request):
-    pass
\ No newline at end of file
+    if request.method=='POST':
+        username = request.POST.get('Username')
+        passwordc = request.POST.get('passwordc')
+        password = request.POST.get('password')
+        fname = request.POST.get('first_name')
+        lname = request.POST.get('last_name')
+        if(password==passwordc):
+            usercheck= auth.authenticate(username=username)
+            if usercheck is not None:
```

Comment:
This isn't a correct way to check whether a username exist. You could've used filter instead.

It raises an error "Username exists" even though the username is new.